src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -316,6 +316,49 @@ public static int GetReplicaCount(this IResource resource)
         }
     }
 
+    /// <summary>
+    /// Get the arguments from the given resource.
+    /// </summary>
+    /// <remarks>
+    /// This method is useful when you want to make sure the arguments are added properly to resources, mostly in test situations.
+    /// This method has asynchronous behavior when arguments were provided from <see cref="IValueProvider"/> otherwise it will be synchronous.
+    /// </remarks>
+    /// <param name="resource">The resource to get the arguments from.</param>
+    /// <returns>The arguments retrieved from the resource.</returns>
+    public static async ValueTask<IReadOnlyList<string>> GetArgumentListAsync(this IResourceWithArgs resource)
```

Comment:
Are there any places in the current code that can be using this new API now? I see a bunch of places where we already have this logic. Especially in tests.

Reply:
Yes, there is. Do you want me to update those codes that use this new API or can I do this in another PR?